index.html

```diff
@@ -90,54 +90,178 @@
 
+The `Sensor` interface is a lightweight object that represents an actual physical sensor. Concrete sensor implementation will need to subclass this.
+
+If no `identifier` is given, the Constructor uses the identifier of the default sensor of that type. This is to the catter for the common case of having a single sensor of a given type on a device.
+
+<pre class="example highlight javascript">
+
+let sensor = new GeolocationSensor();
+sensor.isDefault; // returns true;
+
+</pre>
+
 <pre class="idl">
-[Constructor(SensorInit sensorInitDict), Exposed=(Window,Worker)]
-interface Sensor : EventTarget {
-  static Promise&lt;sequence&lt;DOMString&gt;&gt; getAll();
+[Constructor(optional DOMString identifier, optional SensorInit sensorInitDic), Exposed=(Window,Worker)]
+interface Sensor {
+    readonly attribute DOMString identifier;
+    readonly attribute boolean isDefault;
+    SensorObserver getObserver(optional SensorObserverOptions sensorObserverOptions);
+};
+
+dictionary SensorInit {
+  boolean isDefault;
+};
+</pre>
+
+#### <dfn>Sensor.identifier</dnf>
+
+Returns the identifier of the sensor. This is an opaque DOMString.
+
+#### <dfn>Sensor.isDefault</dnf>
+
+Returns true if the sensor is the default sensor of that type on the device, false otherwise.
+
+#### <dfn>Sensor.getObserver</dnf>
+
+Returns a new <a>SensorObserver</a> object attached to this particular sensor.
+
+<p class=ednote>Not sure if this is really necessary.</p>
+
+<pre class="example highlight">
+let obs = new GeolocationSensor().getObserver({ frequency: 60 });
+obs.onchange = dostuff;
+
+// Equivalent to:
+let sensor = new GeolocationSensor();
+let obs = new SensorObserver(sensor, { frequency: 60 });
+obs.onchange = dostuff;
+</pre>
+
+### The <dfn>SensorObserver</dfn> Interface
+
+The `SensorObserver` interface's role is to observe the changes in a given sensor at regular inetervals and emit events when those value change in a consumer-configurable way.
+
+<pre class="idl">
+[Constructor(Sensor sensor, optional SensorObserverOptions sensorObserverOptions), Exposed=(Window,Worker)]
+interface SensorObserver : EventTarget {
   readonly attribute double frequency;
+  readonly attribute boolean batch;
+  readonly attribute Sensor sensor;
+  readonly attribute TresholdCallback? treshold; 
+  readonly attribute double timeout; 
+  readonly attribute booelan wakeup; 
   readonly attribute SensorReading? value;
+  readonly attribute SensorReading[]? values;
   attribute EventHandler onerror;
   attribute EventHandler onchange;
+  // would that cover all cases?
+  attribute EventHandler oncalibration;
+  // needed?
+  attribute EventHandler onconnect;
+  attribute EventHandler ondisconnect;
 };
 
-dictionary SensorInit {
+dictionary SensorObserverOptions {
   double? frequency;
-  DOMString? identifier;
+  boolean? batch = false;
+  TresholdCallback? treshold;
+  double? timeout;
 };
+
+callback TresholdCallback = boolean (SensorReading currentValue, SensorReading newValue);
```

Comment:
It would need to stop propagation and things which afaik is registration order dependent, no? I need to think about this more (there's an issue for that: #25).